src/sentry/search/events/types.py

```diff
@@ -74,24 +75,34 @@ class EventsResponse(TypedDict):
 
 @dataclass
 class SnubaParams:
-    start: datetime | None
-    end: datetime | None
+    start: datetime | None = None
+    end: datetime | None = None
+    stats_period: str | None = None
     # The None value in this sequence is because the filter params could include that
-    environments: Sequence[Environment | None]
-    projects: Sequence[Project]
-    user: RpcUser | None
-    teams: Sequence[Team]
-    organization: Organization | None
+    environments: Sequence[Environment | None] = field(default_factory=list)
+    projects: Sequence[Project] = field(default_factory=list)
+    user: RpcUser | None = None
+    teams: Sequence[Team] = field(default_factory=list)
+    organization: Organization | None = None
 
     def __post_init__(self) -> None:
         if self.start:
             self.start = self.start.replace(tzinfo=timezone.utc)
         if self.end:
             self.end = self.end.replace(tzinfo=timezone.utc)
+        if self.start is None and self.end is None:
+            self.parse_stats_period()
 
         # Only used in the trend query builder
         self.aliases: dict[str, Alias] | None = {}
 
+    def parse_stats_period(self) -> None:
+        if self.stats_period is not None:
+            self.end = django_timezone.now()
+            from sentry.api.utils import get_datetime_from_stats_period
+
+            self.start = get_datetime_from_stats_period(self.stats_period, self.end)
```

Comment:
Nit: Thoughts on using `get_date_range_from_stats_period` that doesnt require you to set `self.end` here?

Reply:
🤔 those require passing a params object no? I rather not make this more circular than it already is